agent_guard_core/credentials/conjur_secrets_provider.py

```diff
@@ -23,43 +25,99 @@
+                 ext_authn_cred_provider=None):
         super().__init__()
         load_dotenv()
 
-        self._url = os.getenv("CONJUR_APPLIANCE_URL")
-        self._workload_id = os.getenv("CONJUR_AUTHN_LOGIN")
-        self._api_key = os.getenv("CONJUR_AUTHN_API_KEY", "")
-        self._authenticator_id = os.getenv("CONJUR_AUTHENTICATOR_ID")
+        # reference to external authn credential provider function
+        self._ext_authn_cred_provider = ext_authn_cred_provider
+
+        init_error = False
+        # common authn values for all authn endpoints
+        self._url = os.getenv("CONJUR_APPLIANCE_URL", None)
+        if self._url is None:
+            self.logger.error(
+                "ConjurSecretsProvider:__init__(): Required env var CONJUR_APPLIANCE_URL is missing."
+            )
+            init_error = True
+
+        self._authenticator_id = os.getenv("CONJUR_AUTHENTICATOR_ID", None)
+        if self._authenticator_id is None:
+            self.logger.error(
+                "ConjurSecretsProvider:__init__(): Required env var CONJUR_AUTHENTICATOR_ID is missing."
+            )
+            init_error = True
+
+        self._workload_id = os.getenv("CONJUR_AUTHN_LOGIN", None)
+        if self._workload_id is None:
+            self.logger.error(
+                "ConjurSecretsProvider:__init__(): Required env var CONJUR_AUTHN_LOGIN is missing."
+            )
+            init_error = True
+
+        if init_error:
+            raise SecretProviderException(
+                "ConjurSecretsProvider: One or more required environment variables are missing."
+            )
+
         self._account = os.getenv("CONJUR_ACCOUNT", DEFAULT_CONJUR_ACCOUNT)
-        self._region = os.getenv("CONJUR_AUTHN_IAM_REGION", region_name)
-        self._access_token = None
-        self._access_token_expiration = datetime.now()
         self._branch = namespace
         self._secret_name = DEFAULT_SECRET_ID
 
-    def _authenticate_aws(self) -> bool:
+        # Define private vars initialized elsewhere
+        self._access_token = None
+        self._access_token_expiration = datetime.now()
+        self._region = None
+
+    # ---- AWS authentication ----
+    def _authenticate_aws_iam(self) -> bool:
         """
         Authenticates with Conjur using AWS IAM role.
 
-        This function uses AWS IAM credentials to authenticate with Conjur. It signs a request using the STS temporary credentials and then fetches an API token from Conjur.
+        This function uses AWS IAM credentials to authenticate with
+        Conjur. It signs a request using the STS temporary credentials
+        and then fetches an API token from Conjur.
 
         Returns:
             bool: True if the authentication succeeded, False if it failed.
         """
 
-        session = boto3.Session()
-        credentials = session.get_credentials()
-        credentials = credentials.get_frozen_credentials()
+        if self._ext_authn_cred_provider is not None:
```

Comment:
You can also do it like this
```suggestion
        if self._ext_authn_cred_provider:
```